cfme/tests/containers/test_labels.py

```diff
@@ -0,0 +1,230 @@
+    obj_values_list = list(chain.from_iterable([d.values() for d in new_lbls_list_revised]))
+
+    return obj_keys_list, obj_values_list, json_api_payload, obj_api_names
+
+
+def verify_labels(provider):
+    obj_keys_list, obj_values_list, json_api_payload, obj_api_names = check_request_status(provider)
+
+    obj_k = map(str, obj_keys_list)
+    obj_v = map(str, obj_values_list)
+    obj_k_api = [item.lower() for item in obj_k]
+
+    for t_obj in TEST_OBJECTS:
+        navigate_to(t_obj, 'All')
+        tb.select("List View")
+        list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
+        if paginator.page_controls_exist():
+            paginator.results_per_page(100)
+        else:
+            logger.warning(
+                'Unable to increase results per page')
+
+        if t_obj is Image:
+            img_ui = [r for r in list_tbl.rows() if obj_api_names[6] in str(r.id.text)]
+            for img in img_ui:
+                obj = Image(img.name.text, img.tag.text, provider)
+                keys_ui = obj.summary.labels.keys
+                if keys_ui:
+                    vals_ui = []
+                    for key in keys_ui:
+                        element = getattr(obj.summary.labels, key)
+                        vals_ui.append(element.value)
+
+            key_found = [e for e in keys_ui if e in '\n'.join(obj_k_api)]
+            value_found = [e for e in vals_ui if e in '\n'.join(obj_v)]
+
+            key_item = strip_string(key_found)
+            val_item = strip_string(value_found)
+
+            dict_ui = dict(zip(keys_ui, vals_ui))
+            key, value = key_item, val_item
+            assert key in dict_ui and value == dict_ui[key]
+        else:
+            obj_ui = [r.name.text for r in list_tbl.rows()]
+            obj_ui_elem = [x for x in obj_ui if x in obj_api_names]
+
+            for name in obj_ui_elem:
+                obj = t_obj(name, provider)
+                keys_ui = obj.summary.labels.keys
+                if keys_ui:
+                    vals_ui = []
+                    for key in keys_ui:
+                        element = getattr(obj.summary.labels, key)
+                        vals_ui.append(element.value)
+
+            key_found = [e for e in keys_ui if e in '\n'.join(obj_k_api)]
+            value_found = [e for e in vals_ui if e in '\n'.join(obj_v)]
+
+            key_item = strip_string(key_found)
+            val_item = strip_string(value_found)
+
+            dict_ui = dict(zip(keys_ui, vals_ui))
+            key, value = key_item, val_item
+            assert key in dict_ui and value == dict_ui[key]
+
+
+@pytest.mark.polarion('CMP-10572')
+def test_labels(provider):
+    # verify the created labels in CFME
+    navigate_to(provider, 'Details')
+    tb.select(
+        'Configuration',
+        'Refresh items and relationships',
+        invokes_alert=True)
+    sel.handle_alert()
+
+    time.sleep(90)
+    tb.select('Reload Current Display')
```

Comment:
Why are you doing all this refresh before you added the labels? what are you trying to achieve from this refresh?